### PR TITLE
[SYCL] Change uint to uint32_t to avoid build error on Windows

### PR DESCRIPTION
**Pull request by v-klochkov**
Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

**Comment by v-klochkov** [comment]
Vladimir, those other places where uint is used do not cause any errors on Windows.
For commands.h/cpp I had choice: to use cl::sycl::uint or use uint32_t. I chose the 2nd.

**Comment by vladimirlaz** [comment]
> Vladimir, those other places where uint is used do not cause any errors on Windows.
> For commands.h/cpp I had choice: to use cl::sycl::uint or use uint32_t. I chose the 2nd.

Could you explain why we have warning in one place and do not have in the other for the similar code?

**Comment by v-klochkov** [comment]
Errors, not warnings.
Initially I thought it was something wrong with namespaces, but after an additional quick investigation I realized that the alternative fix to my patch is to simply add add '#include <types.hpp>' to commands.h.

It is rather the matter of taste here (cl::sycl::uint vs uint32_t).
uint32_t is used much more often in SYCL sources/headers than 'uint'. So, I would stick to the current 'uint32_t' fix.

**Comment by vladimirlaz** [comment]
> Errors, not warnings.
> Initially I thought it was something wrong with namespaces, but after an additional quick investigation I realized that the alternative fix to my patch is to simply add add '#include <types.hpp>' to commands.h.
> 
> It is rather the matter of taste here (cl::sycl::uint vs uint32_t).
> uint32_t is used much more often in SYCL sources/headers than 'uint'. So, I would stick to the current 'uint32_t' fix.

Thank you for clarification. LGTM.

File: sycl/include/CL/sycl/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ void ExecuteKernelCommand<
   }
 
   if (m_KernelArgs != nullptr) {
-    unsigned ArgumentID = 0;
+    uint32_t ArgumentID = 0;
```
Comment (bader) [inline sycl/include/CL/sycl/detail/scheduler/commands.cpp]: This change seems to be unrelated to the fix.
Suggest reverting.

Comment (v-klochkov) [reply sycl/include/CL/sycl/detail/scheduler/commands.cpp]: This change IS related to the fix.
The fix changed the 1st argument of passGlobalAccessorAsArg() and passLocalAccessorAsArg from uint to uint32_t and ArgumentID is passed to there.

BTW, your comment helped me to find couple places where I did not move the patch properly from WIndows (where it was complete) to Linux (where I also tested the patch and from where uploaded fix for review). Those 2 places are fixed now (changed uint to uint32_t there).

File: sycl/include/CL/sycl/detail/scheduler/commands.h [hunk 2]
```diff
@@ -408,11 +408,11 @@ template <int DimSrc, int DimDest> class CopyCommand : public Command {
 // accessor in lambda function. 'ClKernel' is the kernel. 'HostKernel'
 // is the pointer to the lambda function.
 template <int AccessDimensions, typename KernelType>
-uint passGlobalAccessorAsArg(uint I, int LambdaOffset, cl_kernel ClKernel,
-                             const KernelType &HostKernel);
+uint32_t passGlobalAccessorAsArg(uint I, int LambdaOffset, cl_kernel ClKernel,
```
Comment (bader) [inline sycl/include/CL/sycl/detail/scheduler/commands.h]: Are you going to replace `uint I`?

Comment (v-klochkov) [reply sycl/include/CL/sycl/detail/scheduler/commands.h]: Yes, thank you. Sorry, I did 'git commit' without doing 'git add', that is why uint -> uint32_t for this parameter 'I' was not updated here 20 minutes ago. It is fixed now.
